AttributeError: 'NoneType' object has no attribute 'order_id'

PYTHONPATH=./ pytest .
=========================================== test session starts ===========================================
platform linux -- Python 3.9.2, pytest-7.1.2, pluggy-1.0.0
rootdir: /root/tinkoff-trading-bot, configfile: pytest.ini
plugins: mock-3.7.0, asyncio-0.18.3
asyncio: mode=auto

tests/strategies/interval/backtest/test_on_historical_data.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
app/strategies/interval/IntervalStrategy.py:253: in start
    await self.main_cycle()
app/strategies/interval/IntervalStrategy.py:227: in main_cycle
    await self.validate_stop_loss(last_price)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <app.strategies.interval.IntervalStrategy.IntervalStrategy object at 0x7fbcd416c9d0>
last_price = 12.5

    async def validate_stop_loss(self, last_price: float) -> None:
        """
        Check if stop loss is reached. If yes, then sells all the shares.
        :param last_price: Last price of the instrument.
        """
        positions = (await client.get_portfolio(account_id=self.account_id)).positions
        position = get_position(positions, self.figi)
        if position is None:
            return
        position_price = quotation_to_float(position.average_position_price)
        if position_price < last_price:
            logger.info(f"Stop loss triggered. Last price={last_price} figi={self.figi}")
            try:
                posted_order = await client.post_order(
                    order_id=str(uuid4()),
                    figi=self.figi,
                    direction=ORDER_DIRECTION_SELL,
                    quantity=int(quotation_to_float(position.quantity)),
                    order_type=ORDER_TYPE_MARKET,
                    account_id=self.account_id,
                )
            except Exception as e:
                logger.error(f"Failed to post sell order. figi={self.figi}. {e}")
                return
            asyncio.create_task(
                self.stats_handler.handle_new_order(
>                   order_id=posted_order.order_id, account_id=self.account_id
                )
            )
E           AttributeError: 'NoneType' object has no attribute 'order_id'

app/strategies/interval/IntervalStrategy.py:195: AttributeError

--- a/tests/strategies/interval/backtest/conftest.py
+++ b/tests/strategies/interval/backtest/conftest.py
@@ -1,3 +1,4 @@
+import uuid
 from datetime import timedelta
 from pathlib import Path
 from typing import List
@@ -16,7 +17,7 @@ from tinkoff.invest import (
     PortfolioPosition,
     Quotation,
     OrderDirection,
-    MoneyValue,
+    MoneyValue, PostOrderResponse,
 )
 from tinkoff.invest.caching.cache_settings import MarketDataCacheSettings
 from tinkoff.invest.services import MarketDataCache, Services
@@ -162,6 +163,8 @@ class PortfolioHandler:
             self.positions -= quantity
             self.resources += quantity * last_price - (self.comission * quantity * last_price)
             self.average_price = MoneyValue(units=0, nano=0)
+
+        return PostOrderResponse(order_id=uuid.uuid4().hex)
 
 
 @pytest.fixture(scope="session")

